src/software/amazon/ion/impl/IonReaderTextRawTokensX.java

```diff
@@ -2035,17 +2048,36 @@ protected int load_double_quoted_string(StringBuilder sb, boolean is_clob)
                 }
                 break;
             }
-
+            // if this isn't a clob we need to decode UTF8 and
+            // handle surrogate encoding (otherwise we don't care)
             if (!is_clob) {
+                expectLowSurrogate = check_for_low_surrogate(c, expectLowSurrogate);
+
                 if (IonUTF8.needsSurrogateEncoding(c)) {
                     sb.append(IonUTF8.highSurrogate(c));
                     c = IonUTF8.lowSurrogate(c);
+                } else {
+                    expectLowSurrogate = IonUTF8.isHighSurrogate(c);
                 }
             }
             sb.append((char)c);
         }
     }
 
+    private boolean check_for_low_surrogate(int c, boolean expectLowSurrogate) throws IonException
```

Comment:
Since we already mix camel and snake case in this file, I'd be in favor of going with camel here since it's the more common style in Java.

It also feels kind of strange to have a method return a boolean, but only ever return `false`.

Reply:
A sizeable majority of method names in this file are named with snake case, so I'm going to stick with that approach.

And I agree, the "always false" method idea is a bid odd, but I do like it in this case, as it allows `expectLowSurrogate` to be reset on the same line as each call to `check_for_low_surrogate()`.

Thanks for taking a look!